data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/SearchConfiguration.java

```diff
@@ -10,15 +10,21 @@
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import jakarta.validation.constraints.AssertTrue;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.SearchContextType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+
 import java.util.Map;
 
 public class SearchConfiguration {
 
     private static final ObjectMapper objectMapper = new ObjectMapper();
     private static final Logger LOG = LoggerFactory.getLogger(SearchConfiguration.class);
 
+    // TODO: Should we default this to NONE and remove the version lookup to determine scroll or point-in-time as the default behavior?
```

Comment:
My thought is that this looks like a similar situation to s3.compression - we can set NONE as default and have "AUTO" as an explicit option to do the version-dependent search.

Reply:
That's definitely a consideration. We can think about it more as this feature gets closer to release